scripts/abins/abinsalgorithm.py

```diff
@@ -613,7 +613,13 @@ def write_workspaces_to_ascii(scale: float = 1.0, *, ws_name: str) -> None:
         num_workspaces = mtd[ws_name].getNumberOfEntries()
         for wrk_num in range(num_workspaces):
             wrk = mtd[ws_name].getItem(wrk_num)
-            SaveAscii(InputWorkspace=Scale(wrk, scale, "Multiply"), Filename=wrk.name() + ".dat", Separator="Space", WriteSpectrumID=False)
+            SaveAscii(
+                InputWorkspace=(tmp_workspace := Scale(wrk, scale, "Multiply", StoreInADS=False)),
+                Filename=wrk.name() + ".dat",
+                Separator="Space",
+                WriteSpectrumID=False,
+            )
+            DeleteWorkspace(tmp_workspace)
```

Comment:
The `DeleteWorkspace` algorithm only removes the workspace from the ADS if it exists in the ADS. Seeing as StoreInADS=False, the workspace does not exist in the ADS - and so we shouldn't need to use `DeleteWorkspace`

This should be enough for it to work:
```suggestion
            SaveAscii(
                InputWorkspace=Scale(wrk, scale, "Multiply", StoreInADS=False),
                Filename=wrk.name() + ".dat",
                Separator="Space",
                WriteSpectrumID=False,
            )
```

Reply:
Nice use of the walrus operator - I didn't realise it was in python 3.8!

Reply:
We should do some kind of announcement each time the minimum version requirement changes: "here are the neat Python features you can, and probably should, start using" :smile:

Reply:
@ajjackson is that you volunteering to give a code chat on some nice new toys (features)?

Reply:
I suspect others are well-qualified for this but maybe :wink: